src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,87 @@
 package core.basesyntax;
 
+import java.util.Map;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    int size;
+    Node<K, V>[] table;
+    int capacity = 16;
+    double load_factor = 0.75;
+    int threshold = (int) (capacity * load_factor);
+
+    static class Node<K, V> {
+        final K key;
+        V value;
+        Node<K, V> next;
+
+        public Node(K key, V value) {
+            this.key = key;
+            this.value = value;
+            this.next = null;
+        }
+    }
+
+    public int getIndex(K key) {
+        return key.hashCode() % capacity;
```

Comment:
The `getIndex` method may return a negative index if `key.hashCode()` is negative. To ensure the index is non-negative, you can use `Math.abs(key.hashCode()) % capacity` or handle the negative case separately.